### PR TITLE
refactor: adapt types for newer `@endo/eventual-send`

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

Make Endo master integration pass CI again.

### CODE REVIEW

File: packages/wallet/api/src/lib-wallet.js
Comment (mhofman) [inline packages/wallet/api/src/lib-wallet.js:729]: I really dislike this about tsc promise's type handling.

File: scripts/get-packed-versions.sh
Comment (mhofman) [inline scripts/get-packed-versions.sh:36]: 👍

File: scripts/get-packed-versions.sh
Comment (mhofman) [inline scripts/get-packed-versions.sh:42]: Why not do this unconditionally?

**Review by mhofman — APPROVED** [review]
LGTM, but I'm wondering if we shouldn't just always append the hash of the package to the file?

File: scripts/get-packed-versions.sh
Comment (michaelfig) [reply scripts/get-packed-versions.sh:42]: I'd be happy to make that change.  Will do before landing.

**Pull request merged**
